osvimdriver/openstack/heat/driver.py

```diff
@@ -29,29 +35,75 @@ def create_stack(self, stack_name, heat_template, input_properties=None, files=N
-        create_result = heat_client.stacks.create(stack_name=stack_name, template=heat_template, parameters=input_properties, files=files)
-        stack_id = create_result['stack']['id']
-        logger.debug('Stack with name %s created and assigned id %s', stack_name, stack_id)
-        return stack_id
 
-    def delete_stack(self, stack_id):
+        external_request_id = str(uuid.uuid4())
+
+        reqbody_dict = {"stack_name" : stack_name, "template" : heat_template, "parameters" : input_properties, "files" : files}
+        common._generate_additional_logs(reqbody_dict, 'sent', external_request_id, 'application/json',
+                                       'request', 'http', {'method' : 'post', 'uri' : LOG_URI_PREFIX +'/stacks'}, None)
+        
+        
+        try:  
+            create_result = heat_client.stacks.create(stack_name=stack_name, template=heat_template, parameters=input_properties, files=files)
+            stack_id = create_result['stack']['id']
+            driver_request_id = rd.build_request_id(rd.CREATE_REQUEST_PREFIX, str(stack_id))
+            common._generate_additional_logs(create_result, 'received', external_request_id, 'application/json',
+                                       'response', 'http', {'status_code' : 201,'status_reason_phrase' : 'Created'}, driver_request_id)
+            logger.debug('Stack with name %s created and assigned id %s', stack_name, stack_id)
+            return stack_id,driver_request_id
+        except (heatexc.HTTPNotFound,heatexc.HTTPBadRequest) as e:
+            stack_id = str(uuid.uuid4())
+            driver_request_id = rd.build_request_id(rd.CREATE_REQUEST_PREFIX, str(stack_id))
+            status_reason_phrase = 'Not Found'
+            if  e.code != 404:
+                status_reason_phrase = 'Bad Request'
+            common._generate_additional_logs(e, 'received', external_request_id, 'plain/text',
+                                       'response', 'http', {'status_code' : e.code,'status_reason_phrase' : status_reason_phrase}, driver_request_id)
+            raise e
+
+    def delete_stack(self, stack_id, driver_request_id=None):
         if stack_id is None:
             raise ValueError('stack_id must be provided')
         heat_client = self.__get_heat_client()
         logger.debug('Deleting stack with id %s', stack_id)
         try:
+            external_request_id = str(uuid.uuid4())
+           
+            common._generate_additional_logs('', 'sent', external_request_id, '',
+                                        'request', 'http', {'method':'delete', 'uri' : LOG_URI_PREFIX + '/stacks/' + stack_id}, driver_request_id)
             delete_result = heat_client.stacks.delete(stack_id)
-        except heatexc.HTTPNotFound as e:
+            result = ''
+            if delete_result != None:   
+                result = delete_result
+            common._generate_additional_logs(result, 'received', external_request_id, '',
```

Comment:
Would the content type be '', if the delete_result != None. You may need to handle the other case here for content type.

Reply:
I have made changes.